crates/common/tedge_config_macros/impl/src/query.rs

```diff
@@ -363,9 +381,26 @@ fn generate_string_writers(paths: &[VecDeque<&FieldOrGroup>]) -> TokenStream {
                 parse_quote_spanned! {ty.span()=>
                     WritableKey::#variant_name => self.#(#segments).* = None,
                 },
+                parse_quote_spanned! {ty.span()=>
+                    WritableKey::#variant_name => self.#(#segments).* = <#ty as AppendRemoveItem>::append(
+                        #default_value,
+                        value
+                        .#parse
+                        .#convert_to_field_ty
+                        .map_err(|e| WriteError::ParseValue(Box::new(e)))?),
+                },
+                parse_quote_spanned! {ty.span()=>
+                    WritableKey::#variant_name => self.#(#segments).* = <#ty as AppendRemoveItem>::remove(
+                        #default_value,
```

Comment:
Now we're using it unconditionally, we should call it something like `reader_value` or `current_value`